pkg/make_deb.py

```diff
@@ -308,26 +308,40 @@ def CreateChanges(output,
 
 
 def GetFlagValue(flagvalue, strip=True):
-  """Normalize a flag value.
+  """Converts a raw flag string to a useable value.
 
-  Handles @filename expansions.
+  1. Expand @filename style flags to the content of filename.
+  2. Cope with Python3 strangness of sys.argv.
+     sys.argv is not actually proper str types on Unix with Python3
+     The bytes of the arg are each directly transcribed to the characters of
+     the str. It is actually more complex than that, as described in the docs.
+       https://docs.python.org/3/library/sys.html#sys.argv
+       https://docs.python.org/3/library/os.html#os.fsencode
+       https://www.python.org/dev/peps/pep-0383/
 
   Args:
-    flagvalue: (bytes|str|unicode) A value
-    strip: Strip white space?
+    flagvalue: (str) raw flag value
+    strip: (bool) Strip white space.
 
   Returns:
     Python2: unicode
     Python3: str
   """
   if flagvalue:
     if sys.version_info[0] < 3:
+      # python2 gives us raw bytes in argv.
       flagvalue = flagvalue.decode('utf-8')
-    else:
-      flagvalue = str(flagvalue)
+    # assertion: py2: flagvalue is unicode
+    # assertion: py3: flagvalue is str, but in weird format
     if flagvalue[0] == '@':
+      # Subtle: We do not want to re-encode the value here, because it
+      # is encoded in the right format for file open operations.
       with open(flagvalue[1:], 'rb') as f:
         flagvalue = f.read().decode('utf-8')
+    else:
+      # convert fs specific encoding back to proper unicode.
+      flagvalue = os.fsencode(flagvalue).decode('utf-8')
```

Comment:
`fsencode()` will return the original bytes. If we're on Windows, what encoding will those bytes be? If they're not utf-8, doesn't that mean we'll be mis-decoding `flagvalue`?

Another way to put it: `sys.argv` gives us a sequence of unicode code points as specified by the system. On windows this can actually be interesting characters (>255), while on linux this is always characters in range 0-255. This is because the use of utf-8 in linux is a convention, not a built-in feature understood by the OS. Therefore, unconditionally decoding the bytes as utf-8 is (I think?) incorrect on systems that follow a different convention.